pkg/models/problem.go

```diff
@@ -66,27 +69,36 @@ func (p *StatusProblem) ProblemType() ProblemType {
 	return p.Type
 }
 
+func (p *StatusProblem) ProblemStatus() int {
+	return p.Status
+}
+
 func (p *StatusProblem) ProblemTitle() string {
 	return p.Title
 }
 
 // Respond will render the problem as JSON to the provided ResponseWriter.
 func (p *StatusProblem) Respond(w http.ResponseWriter, r *http.Request) {
+	RespondProblem(p, w, r)
+}
+
+// Respond will render the problem as JSON to the provided ResponseWriter.
+func RespondProblem(problem Problem, w http.ResponseWriter, r *http.Request) {
 	// Respond
 	buf := &bytes.Buffer{}
 	enc := json.NewEncoder(buf)
 	enc.SetEscapeHTML(true)
-	_ = enc.Encode(p)
+	_ = enc.Encode(problem)
 
 	w.Header().Set("Content-Type", ProblemContentType)
-	w.WriteHeader(p.Status)
+	w.WriteHeader(problem.ProblemStatus())
 	_, _ = w.Write(buf.Bytes())
 }
 
 // NewStatusProblem will generate a problem for the provided HTTP status
 // code. The Problem's Status field will be set to match the status argument,
 // and the Title will be set to the default Go status text for that code.
-func NewStatusProblem(ctx context.Context, err error, status int) Problem {
+func NewStatusProblem(ctx context.Context, err error, status int) *StatusProblem {
```

Comment:
why did you change this? (just curious)

Reply:
Check this: https://go.dev/play/p/GGaWrq1Gu3n

The overal issue is that if an interface is composed with a struct, then json marshaling will treat that as a field member as opposed to being part of the same object.

If we want to define per service problems, like this: https://github.com/openmeterio/openmeter/pull/904/files#diff-f49bfff64b9ead64bbdc915228760a76fb213fcbc0474605aef0c0a5f47c8ad5R19 without putting them into models, we need to have access to the *StatusProblem.

Now I could have done a typecast there, but then if something changes in NewStatusProblem's implementation that might break runtime, so it's cleaner to expose the struct instead.